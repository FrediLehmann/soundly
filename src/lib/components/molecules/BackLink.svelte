<script lang="ts">
  import { Link } from '$lib/components/atoms';
  import { ArrowLeft } from '$lib/components/Icons';

  export let href: string;
</script>

<Link
  {href}
  sveltekit:prefetch
  class={`${
    $$props.class ? `${$$props.class} ` : ''
  }flex items-center font-semibold`}>
  <ArrowLeft class="w-5 h-5 mr-2" />
  <span>
    <slot />
  </span>
</Link>
