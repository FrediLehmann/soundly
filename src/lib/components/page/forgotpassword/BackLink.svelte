<script lang="ts" context="module">
  import { Link } from '$lib/components/atoms';
  import { ArrowLeft } from '$lib/Icons';
</script>

<Link href="/signin" class="mbe-4">
  <ArrowLeft width="1.25rem" height="1.25rem" />
  Back
</Link>

<style>
  :global(.mbe-4) {
    margin-block-end: 1rem;
  }
</style>
