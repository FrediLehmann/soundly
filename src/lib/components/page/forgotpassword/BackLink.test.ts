/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom';
import { render } from '@testing-library/svelte';

import BackLink from './BackLink.svelte';

describe('<BackLink /> component', () => {
  test('renders correctly', () => {
    const { component } = render(BackLink);
    expect(component).toMatchSnapshot();
  });
});
