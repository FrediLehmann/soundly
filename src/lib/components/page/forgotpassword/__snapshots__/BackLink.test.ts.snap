// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackLink /> component renders correctly 1`] = `
BackLink {
  "$$": Object {
    "after_update": Array [],
    "before_update": Array [],
    "bound": Object {},
    "callbacks": Object {},
    "context": Map {},
    "ctx": Array [],
    "dirty": Array [
      -1,
    ],
    "fragment": Object {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": Array [
      [Function],
    ],
    "on_disconnect": Array [],
    "on_mount": Array [],
    "props": Object {},
    "root": <div>
      <a
        class="mbe-4 svelte-1u7l9fe"
        href="/signin"
      >
        <svg
          fill="none"
          height="1.25rem"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1.25rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="19"
            x2="5"
            y1="12"
            y2="12"
          />
          <polyline
            points="12 19 5 12 12 5"
          />
        </svg>
        
  Back
      </a>
    </div>,
    "skip_bound": false,
    "update": [Function],
  },
  "$capture_state": [Function],
}
`;
